src/proxy.rs

```diff
@@ -237,6 +252,46 @@ impl RootContext for RootAuthThreescale {
     }
 }
 
+impl HttpAuthThreescale {
+    fn threescale_info_to_metadata(&self, ar: &authrep::AuthRep) -> Result<(), anyhow::Error> {
+        let apps = ar.apps();
+        let service = ar.service();
+        let usages = ar.usages();
+
+        if apps.is_empty() {
+            anyhow::bail!("could not extract application credentials");
+        }
+
+        if apps.len() > 1 {
+            debug!(
+                "found more than one source match for application - going to send {:?}",
+                apps[0]
+            );
+        }
+
+        let mut app_id_key = String::new();
+        let key_header = match &apps[0] {
+            Application::AppId(app_id, app_key) => {
+                app_id_key.push_str(app_id.as_ref());
+                app_id_key.push(':');
+                if let Some(key) = app_key {
+                    app_id_key.push_str(key.as_ref());
+                }
+                vec!["x-3scale-app-id", app_id_key.as_str()]
+            }
+            Application::UserKey(user_key) => vec!["x-3scale-user-key", user_key.as_ref()],
+            Application::OAuthToken(token) => vec!["x-3scale-oauth-token", token.as_ref()],
```

Comment:
This application type is legacy and only exists to support compatibility code that will never even remotely sit close to an Envoy/Proxy-Wasm gateway. In this case I'd just use return an error.